### ISSUE TITLE
Robotlegs plugin ideas

### ISSUE THREAD

**Issue by markstar** [johnlindquist/open-source-plugins#2]
Hi John,

I've seen the videos of your RL plugin it's looking really impressive! I saw your tweet asking for ideas, so here's a couple I had:
- Support Signals based RL projects, including things like the SignalCommandMap and the ApplicationSignalBus.
- Allow users to toggle between Signals based projects and Event based projects.
- Support for creating new projects, with options to download the latest SWCs for popular RL utils like the SignalCommandMap.
- Automate Signal + Command creation with Signal params.
- Automate View + Mediator creation (I think you may be doing this already)
- Give warnings for Signals/Events that have listeners added in onRegister but not removed in onRemove

I'm sure there's more but it's all I can think of from the top of my head.

Keep up the good work! It looks like it'll become an invaluable asset to IntelliJ for RL developers!

Mark

**Comment by joelhooks** [comment]
the "options to download popular SWCs" bullet begs the broader question of how these libraries should be managed. Not really the forum for it, just wanted to write it down as I thought about it ;>

**Comment by nolazybits** [comment]
A simple sorting on the object name ( command,signal,mediator, ... ) in all the map list would be nice also :)
Thx again for this plugin. Real nice :)

**Comment by johnlindquist** [comment]
I'm looking into doing a separate (not specific to Robotlegs) plugin for "Project Generation" which would be part of the "New Module/Project" Wizard process (and as Joel said, we need to find a good way to manage remote swcs. Maybe even leech off of a Maven or Sprouts process that's already set up).

As for Robotlegs code generation, I have a lot of plans :)

**Comment by joelhooks** [comment]
I think you should consider multiple repos instead of one generic one.
If you haven't already.

On Jun 25, 2011, at 20:49, johnlindquist
reply@reply.github.com
wrote:

**Comment by johnlindquist** [comment]
Yeah, still not sure what I think about that since my plugins are going to start sharing classes/Utils/etc and it's just easier to keep it all in one repo.

**Comment by joelhooks** [comment]
Git submodules to the rescue! If it is strictly for you forever, than
sure, but it adds a decent barrier to entry for anybody else to
contribute.

On Jun 25, 2011, at 21:02, johnlindquist
reply@reply.github.com
wrote:

**Comment by nolazybits** [comment]
Hop, as quick message to tell u that I have set up a maven report especially
for actionscript/flex libraries. I've ask stray and they are happy
(hopefully still ok)  to deploy robotleg on it. I'll start soon to
tweet/blog/... about it .feel free to use it for ur projects.
You will need a login that I can create if you give me an email address.

Take care
++
On Jun 26, 2011 1:49 PM, "johnlindquist" <
reply@reply.github.com>
wrote:

**Comment by johnlindquist** [comment]
It's definitely not "strictly for me forever", so I'd like to lower the barrier of entry. I do need to start organizing/cleaning up everything since it started more or less as an experiment and turned out surprisingly well. I'll chat with you more about it off-list later.

**Comment by nolazybits** [comment]
And the link ahah
http://as-artifacts.org
